src/overlays/OverlaidPortal/index.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import { useCallback, useEffect } from 'react';
+import styled from 'styled-components';
+
+import { isClient } from '../../utils';
+import { Portal } from '../Portal';
+
+export interface OverlayProps {
+  className?: string;
+  children: React.ReactNode;
+  opened: boolean;
+  /** overlay 존재 여부 */
+  dimmer?: boolean;
+  /** overlay 색상, 기본 값: rgba(0, 0, 0, 0.72) */
+  overlayColor?: string;
+  closeable?: boolean;
+  onClose?: () => void;
+  /** CSS z-index 속성의 값 */
+  zIndex?: number;
+}
+
+export const OverlaidPortal = React.memo<OverlayProps>(props => {
+  const {
+    className,
+    children,
+    dimmer = true,
+    overlayColor = 'rgba(0, 0, 0, 0.72)',
+    opened,
+    closeable = true,
+    onClose,
+    zIndex = 3000,
+  } = props;
+
+  useEffect(() => {
+    if (isClient() && dimmer) {
+      document.body.style.overflow = opened ? 'hidden' : '';
+    }
+    return () => {
+      if (isClient() && dimmer) {
+        document.body.style.overflow = '';
+      }
+    };
+  }, [opened, dimmer]);
+
+  const handleCloseModal = useCallback(() => {
+    if (onClose) {
+      onClose();
+    }
+  }, [onClose]);
+
+  if (!dimmer) {
+    return <Portal container={document.body}>{children}</Portal>;
+  }
+
+  return (
+    <Portal container={document.body}>
+      <Overlay
+        className={className}
+        zIndex={zIndex}
+        overlayColor={overlayColor}
+        dimmer={dimmer}
+        visible={opened}
+        onClick={closeable ? handleCloseModal : undefined}
```

Comment:
```suggestion
        onClick={closeable ? onClose : undefined}
```

useCallback 안 쓰고 일케 해도 괜찮을 거 같다!

Reply:
조와~!